packages/php-domain/src/PhpDomain/MethodNameService.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace T3Docs\PhpDomain\PhpDomain;
+
+use T3Docs\PhpDomain\Nodes\MethodNameNode;
+
+class MethodNameService
+{
+    private const PATTERN = '/^\s*(\w+)\s*\(\s*(.*?)\s*\)\s*(?::\s*(\w+))?\s*$/';
+
+    public function getMethodName(string $name): MethodNameNode
+    {
+        if (preg_match(self::PATTERN, $name, $matches)) {
```

Comment:
We should log a warning if the pattern does not match.

Reply:
done